stretch_core/stretch_core/stretch_diagnostics.py

```diff
@@ -0,0 +1,121 @@
+import diagnostic_msgs
+
+import diagnostic_updater
+
+import stretch_body.hello_utils as hello_utils  # For get_fleet_id
+
+# Shortcut constants
+ERROR = diagnostic_msgs.msg.DiagnosticStatus.ERROR
+WARN = diagnostic_msgs.msg.DiagnosticStatus.WARN
+OK = diagnostic_msgs.msg.DiagnosticStatus.OK
+
+def add_keyvalues_to_status(status, d, skip_keys=None):
+    """Add all the key value pairs in dictionary d (except those keys in skip_keys) to the diagnostic status."""
+    for k, v in d.items():
+        if skip_keys and k in skip_keys:
+            continue
+        status.add(k, str(v))
+
+def report_stats(status, d, skip_keys=None):
+    """Diagnostic update function that marks the summary as OK and just reports the stats from d."""
+    status.summary(OK, 'OK')
+    add_keyvalues_to_status(status, d, skip_keys)
+    return status
+
+def analyze_stepper(status, device):
+    """Diagnostic update function for device with stepper motor."""
+    if not device.motor.hw_valid:
+        status.summary(ERROR, 'Hardware not present')
+    elif not device.motor.status['pos_calibrated']:
+        status.summary(WARN, 'Hardware not calibrated')
+    elif device.motor.status['runstop_on']:
+        status.summary(WARN, 'Runstop on')
+    else:
+        status.summary(OK, 'Hardware present')
```

Comment:
Maybe we can change this line (and the line on dynamixel) to something like: `Hardware present and calibrated`, which would also inform the user about calibration status.